lib/rouge/lexers/sql.rb

```diff
@@ -83,21 +83,39 @@ def self.keywords
           USAGE USER USER_DEFINED_TYPE_CATALOG USER_DEFINED_TYPE_NAME
           USER_DEFINED_TYPE_SCHEMA USING VACUUM VALID VALIDATOR VALUES
           VARIABLE VERBOSE VERSION VIEW VOLATILE WHEN WHENEVER WHERE
-          WITH WITHOUT WORK WRITE YEAR ZONE
+          WITH WITHOUT WORK WRITE ZONE
         )
       end
 
+      def self.keywords_type
+        # sources:
+        # https://dev.mysql.com/doc/refman/5.7/en/numeric-type-overview.html
+        # https://dev.mysql.com/doc/refman/5.7/en/date-and-time-type-overview.html
+        # https://dev.mysql.com/doc/refman/5.7/en/string-type-overview.html
+        @keywords_type ||= Set.new(%w(
+            ZEROFILL UNSIGNED SIGNED SERIAL BIT TINYINT BOOL BOOLEAN SMALLINT
+            MEDIUMINT INT INTEGER BIGINT DECIMAL DEC NUMERIC FIXED FLOAT DOUBLE
+            PRECISION REAL
+            DATE DATETIME TIMESTAMP TIME YEAR
+            NATIONAL CHAR CHARACTER SET COLLATE NCHAR BYTE
+            VARCHAR VARYING BINARY VARBINARY TINYBLOB TINYTEXT BLOB TEXT
+            MEDIUMBLOB MEDIUMTEXT LONGBLOB LONGTEXT ENUM
+        ))
+      end
+
       state :root do
         rule /\s+/m, Text
         rule /--.*/, Comment::Single
         rule %r(/\*), Comment::Multiline, :multiline_comments
         rule /\d+/, Num::Integer
         rule /'/, Str::Single, :single_string
-        rule /"/, Name::Variable, :double_string
```

Comment:
This is misleading. Single-quoted and Double-quoted strings have different semantics in many databases, such as e.g. Postgres and SQL Server, where only single-quotes strings refer to actual strings, and double-quoted strings refers to database objects. Unless you know you're highlighting SQL for a specific database type, the old behaviour is more accurate. If you have a strong need to treat them the same, I would suggest a compromise would be to add an option for the lexer, with default to the old behaviour.  E.g. possibly a "dialect" option and/or a specific option for this behaviour. Having a :dialect option might be good to allow further fine-tuning, anyway.